### PR TITLE
fix(bankid): fallback to "0.0.0.0" for null ip used for bankid

### PR DESCRIPTION
**Pull request by MasterKenth**
## Explain the changes you’ve made

Use "0.0.0.0" as IP address for bankid if IP is `null`.

## Explain why these changes are made

In certain cases it seems that the IP returned by the network info plugin is `null`. It seems to affect some Android devices. Using `null` results in an error from the BankID API and causes the devices to be unable to login/sign etc.

## Explain your solution

I've added a fallback to "0.0.0.0" when we fetch the IP for auth and sign operations in `BankidService.js`.

## How to test

Concrete example:

1. Checkout this branch
2. Run and verify that you can still login etc. as usual

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [x] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Review by dannil76 — APPROVED** [review]
Looks right to me!

File: source/services/BankidService.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ async function collect(orderRef) {
  * @param {string} ssn A Swedish Social Security Number.
  */
 async function auth(ssn) {
-  const endUserIp = await getIPv4Address();
+  const endUserIp = (await getIPv4Address()) || "0.0.0.0";
```
Comment (karlknowit) [inline source/services/BankidService.js:55]: How about to default to "0.0.0.0" in `getIPv4Address` function instead of having duplicate code that does it?

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/services/BankidService.js (hunk 1 above)
Comment (dannil76) [reply source/services/BankidService.js:55]: Please show how :)

Comment (karlknowit) [reply source/services/BankidService.js:55]: @dannil76  Check function `getIPV4Address` in path `source/helpers/NetworkInfo.ts`

**Optional change**
```js
export default async function getIPv4Address(): Promise<string> {
  const customNetworkInfoModule: CustomNetworkInfo =
    NativeModules.BasicNetworkInfoModule;

  let ipv4Address;

  if (customNetworkInfoModule) {
    ipv4Address = customNetworkInfoModule.GetIPv4Address();
  } else {
    ipv4Address = NetworkInfo.getIPV4Address();
  }

  return ipv4Address || "0.0.0.0";
}

```

Comment (MasterKenth) [reply source/services/BankidService.js:55]: It should not be up to `getIPv4Address` to default to "0.0.0.0" as that is a very BankID-specific fallback value. `getIPv4Address` should do one thing only and that is return the IP.

If duplicate code is the concern, would lifting the fallback logic into a separate function work?

```typescript
async function getIPForBankID() {
  const ip = await getIPv4Address();
  return ip ?? "0.0.0.0";
}
```

Comment (karlknowit) [reply source/services/BankidService.js:55]: Looks good!

**Review by karlknowit — APPROVED** [review]
shitty commit message, but lets get it merged!

**Pull request merged**